doc/source/rllib/core-concepts.rst

```diff
@@ -163,176 +163,162 @@ These batches are wrapped up together in a ``MultiAgentBatch``,
 
 **Train ops** (`train_ops.py <https://github.com/ray-project/ray/blob/master/rllib/execution/train_ops.py>`__):
-These are functions that improve the policy and update workers. The most basic operator, ``TrainOneStep``, take in as input a batch of experiences and emit metrics as output.
-Important operators here include ``TrainOneStep``, ``TrainTFMultiGPU`` (for multi-GPU optimization), ``ComputeGradients`` (to compute gradients without updating the policy), and ``ApplyGradients`` (to apply computed gradients to a policy).
-
-**Replay ops** (`replay_ops.py <https://github.com/ray-project/ray/blob/master/rllib/execution/replay_ops.py>`__):
-The main operator provided here is ``StoreToReplayBuffer``, which can save experiences batches to either a local replay buffer or a set of distributed replay actors.
-It has a counterpart, ``Replay``, that produces a new stream of experiences replayed from one of the aforementioned replay buffers.
-Algorithms that use ``StoreToReplayBuffer`` and ``Replay`` are necessarily composed of *multiple sub-dataflows* (different iterators), that are combined with *concurrency ops*.
+These are functions that improve the policy and update workers. The most basic operator, ``train_one_step``, take in as input a batch of experiences and emit metrics as output.
+For training with gpus, use ``multi_gpu_train_one_step``. These functions use the ``learn_on_batch`` function of rollout workers to complete the training update.
 
-**Concurrency ops** (`concurrency_ops.py <https://github.com/ray-project/ray/blob/master/rllib/execution/concurrency_ops.py>`__):
-The main operator provided here is ``Concurrently``, which composes multiple iterators (dataflows) into a single dataflow by executing them in an interleaved fashion.
-The output can be defined to be the mixture of the two dataflows, or filtered to that of one of the sub-dataflows. It has two modes:
+**Replay Buffers**:
+RLlib provides `a collection <https://github.com/ray-project/ray/tree/master/rllib/utils/replay_buffers>`__ of replay buffers that can be used for storing and sampling experiences.
 
-    - ``round_robin``: Alternate taking items from each input dataflow.
-      This ensures a fixed ratio of computations between, e.g., experience generation and experience replay.
-      The ratio can be adjusted by setting ``round_robin_weights``.
-    - ``async``: Execute each input dataflow as fast as possible without blocking.
-      You might want to use this when, e.g., you want replay to proceed as fast as possible irregardless of how fast experiences are being generated.
+**Concurrency ops**:
+RLlib provides a collection of concurrency ops that can be asynchronous and synchronous operations in the training loop.
```

Comment:
synchronous_parallel_sample can be found in our rollout_ops.py, while concurrency_ops.py contains the parallel iterator related logic. Combined with us not referencing anything here, this does not look clean to me.

Reply:
I have now moved it to parallel requests.py